components/new_post/post_component.tsx

```diff
@@ -0,0 +1,333 @@
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    a11yIndex?: number;
+    isBot: boolean;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+        removePost: (post: Post) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isMobileView: boolean;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const postRef = useRef<HTMLDivElement>(null);
+    const postHeaderRef = useRef<HTMLDivElement>(null);
+    const [hover, setHover] = useState(false);
+    const [a11yActive, setA11y] = useState(false);
+    const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    useEffect(() => {
+        if (postRef.current) {
+            postRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+        }
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, []);
+
+    useEffect(() => {
+        if (hover) {
+            removeKeyboardListeners();
+        }
+
+        if (postRef.current) {
+            postRef.current.removeEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.removeEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+        }
+    }, [hover]);
+
+    const getClassName = (post: Post, isSystemMessage: boolean, isMeMessage: boolean) => {
+        const hovered =
+            fileDropdownOpened;
+        return classNames('a11y__section post post--thread same--root post--comment', {
+            'post--highlight': props.shouldHighlight,
+            'post--editing': props.isPostBeingEdited,
+            'current--user': props.currentUserId === post.user_id,
+            'post--system': isSystemMessage || isMeMessage,
+            'post--compact': props.compactDisplay,
+            'post--hovered': hovered,
+            'same--user': props.isConsecutivePost,
+            'cursor--pointer': alt && !props.channelIsArchived,
+        });
+    };
+
+    const handleAlt = (e: KeyboardEvent) => {
+        if (alt !== e.altKey) {
+            setAlt(e.altKey);
+        }
+    };
+
+    const handleFileDropdownOpened = (open: boolean) => setFileDropdownOpened(open);
+
+    const handleDropdownOpened = (opened: boolean) => {
+        if (props.togglePostMenu) {
+            props.togglePostMenu(opened);
+        }
+    };
+
+    const setsHover = (e: MouseEvent<HTMLDivElement>) => {
```

Comment:
Is there a more descriptive name that we can use for this? Maybe `handleMouseOver` and `handleMouseLeave` to match the corresponding event handlers?